spock-core/src/main/java/org/spockframework/runtime/ParameterizedFeatureNode.java

```diff
@@ -30,30 +38,106 @@ public SpockExecutionContext execute(SpockExecutionContext context, DynamicTestE
     if (testExecutor.getExecutionCount() < 1) {
      throw new SpockExecutionException("Data provider has no data");
     }
+
+    // do not try to aggregate iteration results if they are reported individually
+    if (featureInfo.isReportIterations()) {
+      return context;
+    }
+
+    if (testExecutor.results.containsKey(FAILED)) {
+      List<Throwable> failures = testExecutor.results
+        .get(FAILED)
+        .stream()
+        .filter(Objects::nonNull)
+        .collect(toList());
+
+      switch (failures.size()) {
+        case 0:
+          throw new SpockAssertionError("At least one iteration failed");
+
+        case 1:
+          sneakyThrow(failures.get(0));
+
+        default:
+          MultipleFailuresError multipleFailuresError = new MultipleFailuresError(null, failures);
+          multipleFailuresError.setStackTrace(failures.get(0).getStackTrace());
+          throw multipleFailuresError;
+      }
+    } else if (!testExecutor.results.containsKey(SUCCESSFUL)) {
+      List<Throwable> abortions = testExecutor.results
+        .get(ABORTED)
```

Comment:
Is `!testExecutor.results.containsKey(SUCCESSFUL)` (after filtering out `FAILED`) is always the same as `ABORTED`? E.g. if those with errors are somewhere handled (and in general if there could be something else - now or in the future) that we would like to handle in some other way?

Reply:
Nearly. at this point it is certain there is no failed.
If there are successful and aborted it should be successful.
If there are only aborted it should be aborted.
I added a check in the new method that makes sure there are only aborted and otherwise throws an `AssertionError` as it is a missing case then, have a look.

Reply:
About that check I was thinking about, thanks.